src/common/targetPlatformHelper.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as os from "os";
+import {ErrorHelper} from "../common/error/errorHelper";
+import {HostPlatform} from "../common/hostPlatform";
+import {InternalErrorCode} from "../common/error/internalErrorCode";
+
+/**
+ * Defines the identifiers of all the mobile target platforms React Native supports.
+ */
+export enum TargetPlatformId {
+    ANDROID,
+    IOS
+}
+
+export class TargetPlatformHelper {
+    /**
+     * Return the target platform identifier for a platform with name {platformName}.
+     */
+    public static getTargetPlatformId(platformName: string): TargetPlatformId {
+        switch (platformName.toLowerCase()) {
+            case "android":
+                return TargetPlatformId.ANDROID;
+            case "ios":
+                return TargetPlatformId.IOS;
+            default:
+                throw new Error("The target platform is not supported.");
```

Comment:
I'd interpolate platformName in this string so it's easier for the user to act on this error.